src/plugins/map/MQ2Map.h

```diff
@@ -124,7 +127,7 @@ extern char maphideStr[MAX_STRING];
 extern MQSpawnSearch MapFilterCustom;
 extern MQSpawnSearch MapFilterNamed;
 
-extern MapFilterOption MapFilterOptions[];
+extern std::vector<MapFilterOption> MapFilterOptions;
```

Comment:
This was changed to a vector but the current usage was an unspecified length array with a terminator (invalid entry). I don't see changes to the code that reads it, so maybe this should stay the same as it was?

Reply:
I did review all usage locations when making this change and I think it handles them appropriately.  I figured this was a QOL update to allow for managing them in the future a tad easier.  I'm happy to revert it if you're unsure about the safety of the change though.

Reply:
The implementation of it is supposed to be a static array of options, with a sentinel that terminates. could use std::array instead, but in either case changing it means the terminator is unnecessary, and so that should be changed as well.

Reply:
There's one spot where I look at the terminator still, whereas I could likely swap it out for just looking at the new vector size.  The other use case would be casting.  We have lots of static_cast, and in theory some of those casts could result in the Invalid option, which is used in places to signify that there is not a parent requirement etc.  Would you like me to remove it and exercise this plugin to ensure things still work, or would you like the option of casting to invalid to be left in place?

Reply:
I don't see how casting is relevant to this particular point. Can you show me an example?

Vector shouldn't be used because this is a static data set. it shouldn't be mutated and the number of elements is known at compile time. that makes it a prime candidate for std::array, but i would _not_ make it a vector. array has size(). But knowing the size means we don't need the terminator. The purpose of the terminator was specifically because we didn't know the size.

Reply:
I wanted the vector because it makes it simpler to add new filter options in the future without mucking with a size on init.  But it's a good point that it's constant.  So you're suggesting to move it back to how it was without any modifications?  Or are you asking me to leverage the size and remove the terminator as well?

Reply:
it would be less work to just change it back. I can convert it to an array later.

Reply:
for the example, we use this method:
inline MapFilterOption& GetMapFilterOption(MapFilter Option)
{
    if (Option < MapFilter::All || Option >= MapFilter::Last)
        return MapFilterInvalidOption;

    return MapFilterOptions[static_cast<size_t>(Option)];
}

Reply:
Yeah you wouldn't change that.

Reply:
I'm not saying that you'd remove MapFilterInvalidOption, i'm saying you'd remove the instance of it at the end of the array.

Reply:
I think this is the last unaddressed comment in the PR.  Were you wanting to change it to an array later as mentioned or leave as is?  I can revert a portion of this commit if you prefer it unchanged: https://github.com/SirCabby/macroquest/commit/9668f0e9dd144e119fc692c116ecdada641d0b73

Reply:
just leave it, i'll address it.